File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/data/daos/IBulletDao.java
package com.dkellycollins.triggerfinger.data.daos;

import com.dkellycollins.triggerfinger.data.entity.IBullet;
import com.dkellycollins.triggerfinger.data.model.IVector;

/**
 * Provides the ability to manage bullet data.
 */
public interface IBulletDao extends IActivityDao {

    Iterable<IBullet> retrieve();
    IBullet retrieve(int id);

    int create(int collidableId, IVector velocity);

    void delete(int id);

}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/daos/IBulletDao.kt
package com.dkellycollins.triggerfinger.data.daos

import com.dkellycollins.triggerfinger.data.entity.IBullet
import com.dkellycollins.triggerfinger.data.model.IVector

/**
 * Provides the ability to manage bullet data.
 */
interface IBulletDao : IActivityDao {

    fun retrieve(): Iterable<IBullet>
    fun retrieve(id: Int): IBullet

    fun create(collidableId: Int, velocity: IVector): Int

    fun delete(id: Int)

}
